L120-requiring-cpp17.md

```diff
@@ -0,0 +1,29 @@
+L120: Requiring C++17 in gRPC Core/C++ Library
+----
+* Author(s): veblush
+* Approver: markdroth
+* Status: Draft
+* Implemented in: n/a
+* Last updated: Oct 14, 2024
+* Discussion at: TBD
+
+## Abstract
+
+gRPC starts requiring C++17.
```

Comment:
Do we want to have to keep publishing new gRFCs every time we add support for a new version of C++, or do we just want to publish a gRFC that says "from now on, we'll follow the OSS foundational C++ support policy"?

Reply:
It's a good point. Our plan is to keep up with the OSS foundational C++ support policy so we can have this last gRFC making it clear and skip new gRFC for the next bump.

Reply:
I mentioned the future plan at the bottom. PTAL.

Reply:
If this is actually our plan going forward, then I think we should re-title this gRFC -- it's less about C++17 specifically than it is about the fact that we're now going to follow the OSS foundational C++ support guidelines.

Maybe this warrants a brief discussion at our Tuesday design review meeting?